lib/services/storageManagement2/lib/operations/storageAccounts.js

```diff
@@ -39,10 +41,42 @@ function StorageAccounts(client) {
  * @param {object} [options.customHeaders] Headers that will be added to the
  * request
  *
- * @param {function} callback
+ * @returns {Promise} A promise is returned
+ * @resolve {HttpOperationResponse<CheckNameAvailabilityResult>} - The deserialized result object.
+ * @reject {Error} - The error object.
+ */
+StorageAccounts.prototype.checkNameAvailabilityWithHttpOperationResponse = function (name, options) {
```

Comment:
The callback used to provide the (err, result, request, response). If someone is interested in it then they can access them.
In Promise land, we can only return one object when resolving the Promise. Hence added this method to return a wrapper object that provides everything.

The standard method below will return a Promise that will resolve with the result (deserialized object).